server/service/packs.go

```diff
@@ -23,26 +25,71 @@ type packResponse struct {
 	TeamIDs  []uint `json:"team_ids"`
 }
 
+func userIsGitOpsOnly(ctx context.Context) (bool, error) {
+	vc, ok := viewer.FromContext(ctx)
+	if !ok {
+		return false, fleet.ErrNoContext
+	}
+	if vc.User == nil {
+		return false, errors.New("missing user in context")
+	}
+	if vc.User.GlobalRole != nil {
+		return *vc.User.GlobalRole == fleet.RoleGitOps, nil
+	}
+	if len(vc.User.Teams) == 0 {
+		return false, errors.New("user has no roles")
+	}
+	for _, teamRole := range vc.User.Teams {
+		if teamRole.Role != fleet.RoleGitOps {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
 func packResponseForPack(ctx context.Context, svc fleet.Service, pack fleet.Pack) (*packResponse, error) {
 	opts := fleet.ListOptions{}
 	queries, err := svc.GetScheduledQueriesInPack(ctx, pack.ID, opts)
 	if err != nil {
 		return nil, err
 	}
 
+	totalHostsCount := uint(0)
+
 	hostMetrics, err := svc.CountHostsInTargets(
 		ctx,
 		nil,
-		fleet.HostTargets{HostIDs: pack.HostIDs, LabelIDs: pack.LabelIDs, TeamIDs: pack.TeamIDs},
+		fleet.HostTargets{
+			HostIDs:  pack.HostIDs,
+			LabelIDs: pack.LabelIDs,
+			TeamIDs:  pack.TeamIDs,
+		},
 	)
 	if err != nil {
-		return nil, err
+		var authErr *authz.Forbidden
+		if !errors.As(err, &authErr) {
+			return nil, err
+		}
+		// Some users (e.g. gitops) are not able to read targets, thus
+		// we do not fail when gathering the total host count to not fail
+		// write packs request.
+		ok, gerr := userIsGitOpsOnly(ctx)
```

Comment:
nit:  Consider renaming ok --> isGitOps ?

Reply:
I'll leave as is if that's ok (a user can be gitops on a team, and observer in another team, hence the "only" at the end)